interfaces/mount/spec.go

```diff
@@ -150,6 +150,30 @@ func (spec *Specification) AddExtraLayouts(layouts []snap.Layout) {
 	}
 }
 
+// createEnsureDirMountEntry creates a mount entry from the given ensure directory spec
+// that instruct snap-update-ns to create missing directories if required.
+func createEnsureDirMountEntry(ensureDirSpec *interfaces.EnsureDirSpec) osutil.MountEntry {
+	var entry osutil.MountEntry
+	entry.Options = []string{osutil.XSnapdKindEnsureDir(), osutil.XSnapdMustExistDir(ensureDirSpec.MustExistDir)}
+	entry.Dir = ensureDirSpec.EnsureDir
+
+	return entry
+}
+
+// AddEnsureDirs adds mount entries to ensure the existence
+// of directories according to the given ensure directory specs.
+func (spec *Specification) AddEnsureDirs(ensureDirSpecs []*interfaces.EnsureDirSpec) {
+	// Walk the path specs in deterministic order, by EnsureDir (the mount point).
+	sort.Slice(ensureDirSpecs, func(i, j int) bool {
```

Comment:
Why do these need to be in a deterministic order?

Reply:
Here I followed `AddLayout` assuming it is useful for testing, or an optimisation of some sort, but at a glance it is not clear this do much more than provide an consistent easy to follow order for testing and inspecting mount file.